PocketKit/Sources/PocketKit/Home/HomeViewModel.swift

```diff
@@ -554,94 +532,96 @@ extension HomeViewModel {
+    }
+
+    private func archive(_ savedItem: SavedItem, at indexPath: IndexPath) {
+        self.source.archive(item: savedItem)
+        tracker.track(event: Events.Home.RecentSavesCardArchive(url: savedItem.url, positionInList: indexPath.item))
     }
 }
 
 // MARK: - Cell Lifecycle
 extension HomeViewModel {
     func willDisplay(_ cell: HomeViewModel.Cell, at indexPath: IndexPath) {
         switch cell {
-        case .loading, .recentSaves, .offline:
+        case .loading, .offline:
+            return
+        case .recentSaves(let objectID):
+            guard let savedItem = source.mainContext.object(with: objectID) as? SavedItem else {
+                Log.breadcrumb(category: "home", level: .debug, message: "Could not turn recent save into Saved Item from objectID: \(String(describing: objectID))")
+                Log.capture(message: "SavedItem is null on willDisplay Home Recent Saves")
+                return
+            }
+            tracker.track(event: Events.Home.RecentSavesCardImpression(url: savedItem.url, positionInList: indexPath.item))
             return
         case .recommendationHero(let objectID), .recommendationCarousel(let objectID):
             guard let recommendation = source.mainContext.object(with: objectID) as? Recommendation else {
+                Log.breadcrumb(category: "home", level: .debug, message: "Could not turn recomendation into Recommendation from objectID: \(String(describing: objectID))")
+                Log.capture(message: "Recommendation is null on willDisplay Home Recommendation")
                 return
             }
 
-            tracker.track(
-                event: ImpressionEvent(component: .content, requirement: .instant),
-                contexts(for: recommendation, at: indexPath)
-            )
+            guard
+                let item = recommendation.item,
+                let slate = recommendation.slate,
+                let slateLineup = slate.slateLineup
+            else {
+                Log.capture(message: "Tried to display recommendation without an associated item, slate and slatelineup, not logging analytics")
+                return
+            }
+
+            tracker.track(event: Events.Home.SlateArticleImpression(url: item.givenURL, positionInList: indexPath.item, slateId: slate.remoteID, slateRequestId: slate.requestID, slateExperimentId: slate.experimentID, slateIndex: indexPath.section, slateLineupId: slateLineup.remoteID, slateLineupRequestId: slateLineup.requestID, slateLineupExperimentId: slateLineup.experimentID, recommendationId: recommendation.remoteID))
```

Comment:
nit: this seems to have a lot of parameters imo. would it be better to just to pass in `slate`, `slateLinup` instead?

Reply:
I'm inclined to leave this right now. I agree with you here, but I really want our analytics explicit about what they need until we see more patterns and things emerge, and they don't need everything in that object right now.